desc/coils.py

```diff
@@ -1247,6 +1247,8 @@ def __add__(self, other):
             return CoilSet(*self.coils, *other.coils)
         if isinstance(other, (list, tuple)):
             return CoilSet(*self.coils, *other)
+        elif isinstance(other, (_MagneticField)):
```

Comment:
the base magnetic field class already has logic like this in its add method, i think the preferred way to handle this is
```python
else:
    return NotImplemented
```

see https://docs.python.org/3/library/constants.html#NotImplemented

Reply:
I guess the reason I made this change was so that you could add something like a `ToroidalMagneticField` and a `CoilSet` together to return a `SumMagneticField` of the two objects,  I found that without this change, it resulted in the NotImplementedError, when this seems like something we should be able to do. 

Maybe I am misunderstanding but when a subclass re-implements a method, that method takes precedence over the superclass implementation right?

Reply:
#847 addressed this